src/sentry/search/snuba/backend.py

```diff
@@ -492,14 +492,27 @@ def query(
         else:
             retention_window_start = None
 
-        group_queryset = self._build_group_queryset(
-            projects=projects,
-            environments=environments,
-            search_filters=search_filters,
-            retention_window_start=retention_window_start,
-            date_from=date_from,
-            date_to=date_to,
-        )
+        if use_group_snuba_dataset:
+            # just use the basic group initialization query which prevents us from
+            # returning groups that are pending deletion or merge
+            # this query is only used after we query snuba to filter out groups we don't want
+            group_queryset = Group.objects.filter(project__in=projects).exclude(
+                status__in=[
+                    GroupStatus.PENDING_DELETION,
+                    GroupStatus.DELETION_IN_PROGRESS,
+                    GroupStatus.PENDING_MERGE,
+                ]
+            )
```

Comment:
I realize that without the above change, we actually end up letting Postgres do the filtering based because after Snuba gets the results, we take the group ids from snuba and plug it into the Postgres based on this `group_queryset`